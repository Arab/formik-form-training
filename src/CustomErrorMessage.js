import React from "react";
const CustomErrorMessage = ({ dataTestId, children }) => (
  <div data-testid={dataTestId} className="error">
    {children}
  </div>
);

export default CustomErrorMessage;
