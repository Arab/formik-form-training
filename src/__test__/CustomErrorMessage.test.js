import React from "react";
import CustomErrorMessage from "../components/CustomErrorMessage";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
  wait,
  prettyDOM
} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import selectEvent from "react-select-event";
import userEvent from "@testing-library/user-event";

const defaultProps = {
  dataTestId: "error-default",
  children: "this is default error message"
};

afterEach(cleanup);

test("renders without crashing", () => {
  const { container, rerender } = render(
    <CustomErrorMessage {...defaultProps} />
  );
  expect(container).toMatchSnapshot();
  rerender(
    <CustomErrorMessage {...defaultProps} dataTestId="error-some-new-error" />
  );
  // console.log(prettyDOM(container));
});
