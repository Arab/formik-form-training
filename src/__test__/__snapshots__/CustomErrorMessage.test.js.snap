// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="error"
    data-testid="error-default"
  >
    this is default error message
  </div>
</div>
`;
